include/modem/location.h

```diff
@@ -161,14 +161,56 @@ struct location_datetime {
 struct location_data_details_gnss {
 	/** Number of satellites tracked at the time of event. */
 	uint8_t satellites_tracked;
+	/** Number of satellites used for a fix. */
+	uint8_t satellites_used;
 	/** PVT data. */
 	struct nrf_modem_gnss_pvt_data_frame pvt_data;
+	/**
+	 * GNSS search time in milliseconds.
+	 *
+	 * This is the time since last start of GNSS operation to the fix or timeout
+	 * including any time LTE has blocked GNSS.
+	 *
+	 * @ref pvt_data has nrf_modem_gnss_pvt_data_frame.execution_time which indicates
+	 * cumulative GNSS execution time since last start excluding any time LTE has blocked
+	 * GNSS.
+	 */
+	uint32_t search_time;
 };
 
-/** Location details. */
+/** Location details for cellular. */
+struct location_data_details_cellular {
+	/* Number of neighbor cells. */
+	uint8_t ncells_count;
+	/* Number of GCI (surrounding) cells. */
+	uint8_t gci_cells_count;
+};
+
+/** Location details for Wi-Fi. */
+struct location_data_details_wifi {
+	/** Number of Wi-Fi APs. */
+	uint16_t ap_count;
+};
+
+/**
+ * Location details.
+ *
+ * Only one of the child structures is filled most of the time depending on the method
+ * found in @ref location_event_data.method member of the parent structure.
+ * There are cases when cellular and Wi-Fi data are combined into a single cloud request
+ * in which case data details for both methods are filled.
+ */
 struct location_data_details {
```

Comment:
There is no time member at the moment. `location_data_details_gnss.search_time` exists which is not the whole time method implementation took but only from `nrf_modem_gnss_start()` function call.
I don't have strong opinion whether other time values are needed as memfault will calculate some session time values based on the events that we send.

Reply:
I've added `method_search_time` here